geopandas/base.py

```diff
@@ -100,6 +100,16 @@ def _generate_sindex(self):
             _sindex = sindex_cls(self.geometry)
             if not _sindex.is_empty:
                 self._sindex = _sindex
+            else:
+                warn(
+                    "Generated spatial index is empty and returned `None`. "
+                    "Future versions of GeoPandas will return zero-length spatial "
+                    "index instead of `None`. Use `len(gdf.sindex) > 0` "
+                    "or `if gdf.sindex` instead of `if gd.sindex is not None` "
+                    "to check for empty spatial indexes.",
+                    FutureWarning,
```

Comment:
```suggestion
                    FutureWarning,                   
                    stacklevel=3,
```

Reply:
I suppose this points to `sjoin` or whoever tried to access the `sindex` attribute? I'm fine with that.

Reply:
It will point to `GeoSeries.sindex`, at least that is what I got when I tried `gpd.GeoSeries().sindex`. Same for gdf.

Reply:
I tried:
```python3
from geopandas import GeoSeries
GeoSeries().sindex
```
Which pointed to `__main__`. So I guess it depends on how you import things. In any case, not a big deal, I think we should just leave it like this and not spend more time on it.